### PR TITLE
Angular: update Angular Change Detection Specifics section

### PR DESCRIPTION
**Pull request by sergepilipchuk**

### CODE REVIEW

File: concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md [hunk 1]
```diff
@@ -2,65 +2,189 @@ Angular activates change detection on each property change. This approach backfi
...
+        PipeTransform,
+    } from '@angular/core';
+
+    enum GridColumnsOption {
+        PersonalDetails = 0,
+        JobDetails = 1,
+    }
+
+    @Pipe({name: 'columnsPipe'})
+
```
Comment (williamvinogradov) [inline concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md]: Let's delete this empty line:
```suggestion
```

File: concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md [hunk 2]
```diff
@@ -2,65 +2,189 @@ Angular activates change detection on each property change. This approach backfi
...
+    }
+
+    @Component({
+        selector: 'app-component',
+        template: `
+            <dx-data-grid
+                [columns]="columnsOption | columnsPipe">
+            </dx-data-grid>
+        `,
+    })
```
Comment (williamvinogradov) [inline concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md]: Let's delete this empty line:
```suggestion
```

File: concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md [hunk 3]
```diff
@@ -2,65 +2,189 @@ Angular activates change detection on each property change. This approach backfi
...
+            PipeTransform, 
+        } from '@angular/core';
+        
+        enum GridColumnsOption {
+            PersonalDetails = 0,
+            JobDetails = 1,
+        }
+
+        @Pipe({ name: 'apply' })
+
```
Comment (williamvinogradov) [inline concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md]: Let's remove this empty line:
```suggestion
```

File: concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md [hunk 4]
```diff
@@ -2,65 +2,189 @@ Angular activates change detection on each property change. This approach backfi
...
+            selector: 'app-component',
             template: `
                 <dx-data-grid
-                    [columns]="getColumns()">
+                    [columns]="getColumns | apply: columnsOption">
                 </dx-data-grid>
             `,
-            styleUrls: ['./app.component.css']
         })
+
```
Comment (williamvinogradov) [inline concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md]: Let's remove this empty line:
```suggestion
```

File: concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md [hunk 5]
```diff
@@ -2,65 +2,185 @@ Angular activates change detection on each property change. This approach backfi
```
Comment (albertov05) [inline concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md:2]: On line 1 (above), please change "approach" to "technique".

File: concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md [hunk 6]
```diff
@@ -2,65 +2,185 @@ Angular activates change detection on each property change. This approach backfi
...
 
 1. The `getColumns()` function returns an array.
-1. The array gets assigned to the `columns` property. This activates change detection.
-1. When comparing the old and new `columns` values, the change detection mechanism calls `getColumns()` again and receives a different array.
-1. The array gets assigned to the `columns` property, and the cycle repeats infinitely.
+2. The array gets assigned to the `columns` property. This activates change detection.
+3. When comparing the old and new `columns` values, the change detection mechanism calls `getColumns()` again and receives a different array.
+4. The array gets assigned to the `columns` property, and the cycle repeats infinitely.
+
+To workaround this behavior, implement one of the following approaches:
```
Comment (albertov05) [inline concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md]:
```suggestion
To work around this behavior, implement one of the following techniques:
```

File: concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md [hunk 7]
```diff
@@ -2,65 +2,185 @@ Angular activates change detection on each property change. This approach backfi
...
+            ],
+            [GridColumnsOption.JobDetails]: [
+                'name',
+                'position',
+                'salary',
+            ],
+        };
+    }
+
+**Rewrite a component’s class method to the custom [Angular pipe](https://angular.io/guide/pipes-custom-data-trans).**
```
Comment (albertov05) [inline concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md]:
```suggestion
**Rewrite a component’s class method to a custom [Angular pipe](https://angular.io/guide/pipes-custom-data-trans).**
```

File: concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md [hunk 8]
```diff
@@ -2,65 +2,185 @@ Angular activates change detection on each property change. This approach backfi
...
+                [columns]="columnsOption | columnsPipe">
+            </dx-data-grid>
+        `,
+    })
+    export class AppComponent {
+        columnsOption = GridColumnsOption.PersonalDetails;
+    } 
 
-- Return an object reference, not a new object, from the function:
+**Cache a returned value of the component’s method (for example, you can use universal apply pipe).**
```
Comment (albertov05) [inline concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md]:
```suggestion
**Cache the return value of the component’s method (for example, you can use a universal apply pipe).**
```

File: concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md [hunk 9]
```diff
@@ -2,65 +2,185 @@ Angular activates change detection on each property change. This approach backfi
...
-
-        <!-- tab: app.component.ts -->
-        import { Component, ChangeDetectionStrategy } from '@angular/core';
-
-        @Component({
-            // ...
-            changeDetection: ChangeDetectionStrategy.OnPush
-        })
-        // ...
+[note]We recommend you to use an arrow function with this approach.
```
Comment (albertov05) [inline concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md]:
```suggestion
[note]We recommend that you use an arrow function with this technique.
```

File: concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md (hunk 5 above)
Comment (sergepilipchuk) [reply concepts/40 Angular Components/20 Component Configuration Syntax/85 Angular Change Detection Specifics.md:2]: Ok

**Pull request merged**